lib/shared_widgets/video_player.dart

```diff
@@ -0,0 +1,143 @@
+import 'dart:async';
+
+import 'package:flick_video_player/flick_video_player.dart';
+import 'package:flutter/material.dart';
+import 'package:video_player/video_player.dart';
+
+class CourseVideoPlayer extends StatefulWidget {
+  final String videoPath;
+
+  const CourseVideoPlayer({Key? key, required this.videoPath})
+      : super(key: key);
+
+  @override
+  State<CourseVideoPlayer> createState() => _CourseVideoPlayerState();
+}
+
+class _CourseVideoPlayerState extends State<CourseVideoPlayer> {
+  late VideoPlayerController _controller;
+  late Future<void> _initializeVideoPlayerFuture;
+  late FlickManager flickManager;
+
+  // bool _onTouch = false;
+
+  // late Timer _timer;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = VideoPlayerController.network(widget.videoPath);
+    _initializeVideoPlayerFuture = _controller.initialize();
+    flickManager = FlickManager(videoPlayerController: _controller);
+
+    // _timer = Timer.periodic(
+    //     const Duration(milliseconds: 100), (_) {
+    //   setState(() {
+    //     _onTouch = _onTouch;
+    //   });
+    // });
+    _controller.setLooping(true);
+    _controller.initialize().then((_) {
+      setState(() {});
+    });
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    flickManager.dispose();
+    // _timer.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: _initializeVideoPlayerFuture,
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.done) {
+          // If the VideoPlayerController has finished initialization, use
+          // the data it provides to limit the aspect ratio of the video.
+
+          return AspectRatio(
+            aspectRatio: _controller.value.aspectRatio,
+            // Use the VideoPlayer widget to display the video.
+            child: FlickVideoPlayer(flickManager: flickManager),
+          );
+        } else {
+          // If the VideoPlayerController is still initializing, show a
+          // loading spinner.
+          return const Center(
+            child: CircularProgressIndicator(),
+          );
+        }
+      },
+    );
+  }
+//   return _controller.value.isInitialized
```

Comment:
That's a lot of commented out code, does it mean you're still working on this functionality?

Reply:
It will be deleted, no I'm not, the video player works, you can't test it with an emulator on iOS you need a real device, this is normal on iOS you can Google it.